src/main/java/com/spring/app/services/mappers/IntegerResultMapper.java

```diff
@@ -19,15 +20,28 @@
     @Override
     public Integer map(List<Object> objects) throws DataNotFoundException {
         if (objects != null && !objects.isEmpty() && objects.get(INDEX) != null) {
-            Object object = objects.get(INDEX);
-            if (object instanceof BigDecimal) {
-                return ((BigDecimal) object).intValue();
-            }
-            if (object instanceof Double) {
-                return ((Double) object).intValue();
-            }
-            throw new DataNotFoundException("The received value is incorrect.");
+            return getInteger(objects.get(INDEX));
         }
         throw new DataNotFoundException("Data not found.");
     }
+
+    /**
+     * Resolver for an object that should contains an integer value
+     *
+     * @param object - input
+     * @return converted object
+     */
+    public static Integer getInteger(Object object) {
+        if (object == null) {
+            throw new DataNotFoundException("Data not found.");
+        }
+        if (object instanceof BigDecimal) {
+            return ((BigDecimal) object).intValue();
```

Comment:
code dublicate

Reply:
I don't know how to resolve this

Reply:
i want to delete all this simple result mappers